Проект Filmorate.

Bikochu opened the pull request:
Создать проект Filmorate.

Bikochu committed:
Создать проект Filmorate.

Bikochu committed:
Создать проект Filmorate.

Bikochu committed:
Поменять версию Spring на 2.7.8

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,89 @@
+
+    @PostMapping()
+    public Film addFilm(@Valid @RequestBody Film film) throws ValidationException {
+        if (film.getName() == null || film.getName().isBlank()) {
```
Для валидации лучше использовать что-то одно: или аннотации, или реализовать вручную. Дублировать функционал не стоит, иначе при внесении изменений придется вносить их в нескольких местах.

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,89 @@
+        if (!films.containsKey(film.getId())) {
+            id++;
+            film.setId(id);
+            log.info("Добавили фильму ID: {} .",id);
```
Код нужно отформатировать. В нескольких местах отсутствует пробел между запятой и следующей за ней переменной. Например:

```java
// Wrong
log.info("Добавили фильму ID: {} .",id);
// Correct
log.info("Добавили фильму ID: {} .", id);
```

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    @PutMapping()
+    public Film updateFilm (@Valid @RequestBody Film film) throws ValidationException {
```
Между именем метода и скобкой не должно быть пробела:

```java
// Wrong
public Film updateFilm (@Valid @RequestBody Film film) throws ValidationException {
// Correct
public Film updateFilm(@Valid @RequestBody Film film) throws ValidationException {
```

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,89 @@
+
+    @PutMapping()
+    public Film updateFilm (@Valid @RequestBody Film film) throws ValidationException {
+        if (films.containsKey(film.getId())) {
```
В таких случаях лучше обходиться без `else`. Пользуемся тем, что выброшенное исключение прерывает выполнение метода и сокращаем вложенность:

```java
if (!films.containsKey(film.getId()) {
    log.error("Номер фильма в коллекции не существует.");
    throw new ValidationException("Такого фильма не существует.");
}
films.put(film.getId(), film);
...
```

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,89 @@
+            }
+            log.info("Обновляем пользоватля {} под номером {} в коллекции.",user.getLogin(),user.getId());
+            users.put(user.getId(), user);
+        } else {
```
Можно обойтись без `else`, также как в методе `updateFilm`.

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,89 @@
+        return user;
+    }
+
+    public void setId(int id) {
```
Метод лучше сделать приватным.

ivanrusanov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,89 @@
+        return film;
+    }
+
+    public void setId(int id) {
```
Метод лучше сделать приватным.

Bikochu committed:
Изменил Тесты.
Оставил валидацию через аннотации.
Создал собственную аннотацию для releaseDate.

Bikochu committed:
Убрал закомментированные строки.
